plutus-ledger-constraints/src/Ledger/Constraints/TxConstraints.hs

```diff
@@ -52,6 +52,20 @@ import Data.Maybe (fromMaybe)
 import Prelude qualified as Haskell
 import Prettyprinter.Render.String (renderShowS)
 
+-- | How tx out datum are embedded in a a Tx
+data OutDatum = Inline Datum | Hashed Datum
+    deriving stock (Haskell.Show, Generic, Haskell.Eq)
+    deriving anyclass (ToJSON, FromJSON)
```

Comment:
Instead of defining our own datatype, how about simply reusing `TxOutDatum` from `cardano-api`? It provides ALL the constructors that we need to support, so might as well use it now.

Reply:
`TxOutDatum` doesn't distinguish the case where a `Datum` is mandatory from the case where it's optional. I can switch to it, but we lose a bit of type-safety.

Reply:
Good point :)

Reply:
Add a comment explaining this rationale?

Reply:
Sure, will do.